### DISCUSSION TITLE
Error querying the database: FATAL: sorry, too many clients already

### DISCUSSION THREAD

**Discussion by r-tomiyama** [prisma/prisma#24656] [Q&A]
### Question

When I run tests with Jest, I get the error "too many clients already."

I used Jest to create a test that starts up and connects to a PostgreSQL database.

The test runs successfully when run individually, but when I run multiple tests together, I get the error "too many clients already."

It seems that multiple PrismaClients are being created.

In the application code, I use InversifyJS to generate only one PrismaClient, but I don't know how to generate only one in the test code as well.

How can I create only one PrismaClient when running tests with Jest?

### How to reproduce (optional)

```typescript
import { Container } from "inversify";
import { PrismaClient } from "prisma";

const container = new Container();

container
  .bind(TYPES.PrismaClient)
  .toDynamicValue(() => new PrismaClient())
  .inSingletonScope();
```

```typescript
const prismaClient = container.get(TYPES.PrismaClient);

describe("Hoge", () => {
  beforeEach(async () => {
    // insert seed data
  });

  afterEach(async () => {
    // clean up DB
  });

  it("fuga", async () => {
    const results = await prismaClient.hoges.findFirst({ where: conditions });

    expect(results).toEqual(expects);
  });

  and more ..
});
```

**Comment by ludralph** [comment]
Hi @r-tomiyama 👋 

From your code, it looks like you are creating a singleton. You confirmed that the test is successful when run individually but fails when run together. Can you try having the container setup in a different file say `container.js` and export it to use in the test. For example:
```js
// container.js

import { Container } from "inversify";
import { PrismaClient } from "@prisma/client";

const container = new Container();

container
  .bind(TYPES.PrismaClient)
  .toDynamicValue(() => new PrismaClient())
  .inSingletonScope();

export default container;
```
and then the test

```js
import container from './container';
const prismaClient = container.get(TYPES.PrismaClient);

describe("Hoge", () => {
  // ... rest of the code remains the same
  it("fuga", async () => {
    const results = await prismaClient.hoges.findFirst({ where: conditions });
    expect(results).toEqual(expects);
  });
  // ... other tests
});
```

Do let us know if this helps.

**Reply by r-tomiyama** [reply]
Thanks, @ludralph !

My sample code was insufficient.
I was exporting and importing container as you have written.

However, I got an error.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
